examples/middleware/middleware.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/config"
+	"github.com/stackitcloud/stackit-sdk-go/services/argus"
+)
+
+// RequestCapturer is a middleware that prints the request and a status it receives.
+func RequestCapturer(status string) config.Middleware {
+	return func(rt http.RoundTripper) http.RoundTripper {
+		return &roundTripperWithCapture{rt, status}
+	}
+}
+
+// roundTripperWithCapture is a custom round tripper that prints the request and a status it receives.
+type roundTripperWithCapture struct {
+	transport http.RoundTripper
+	status    string
+}
+
+func (rt roundTripperWithCapture) RoundTrip(req *http.Request) (*http.Response, error) {
+	returnStr := fmt.Sprintf("%s %s", req.Method, req.URL.String())
+
+	fmt.Println("Captured request:", returnStr)
+	fmt.Println("Status:", rt.status)
+
+	// Proceed with the original round trip
+	// This step is important to preserve the original behavior of the client
+	return rt.transport.RoundTrip(req)
```

Comment:
Even cooler would be something like:

```
func (rt roundTripperWithCapture) RoundTrip(req *http.Request) (*http.Response, error) {
	returnStr := fmt.Sprintf("%s %s", req.Method, req.URL.String())

	fmt.Println("Captured request:", returnStr)

        // Proceed with the original round trip
	// This step is important to preserve the original behavior of the client
	resp, err := rt.transport.RoundTrip(req)
	fmt.Println("Status:", resp.Status)

	return resp, err
}
```

In this case it would only make sense to add one middleware though :thinking:
And I like your example because it allows to check the order of execution

Reply:
Something like what I just commited? It shows the functionality of capturing the response and maintains the demonstration of the order of execution

Reply:
yes, now it's a really complete example :wink: